File: Programmers/유석환/Q42862.swift
import Foundation

func solution(_ n:Int, _ lost:[Int], _ reserve:[Int]) -> Int {
    var students = [-1] + [Int](repeating: 1, count: n) + [-1]
    var result = 0
    
    for item in lost {
        students[item] -= 1
    }
    
    for item in reserve {
        students[item] += 1
    }
    
    for i in 1...n {
        if students[i] == 0 {
            if (students[i - 1] == 2 && students[i + 1] < 2) {
                students[i - 1] -= 1
                students[i] += 1
                continue
            }
            
            if (students[i - 1] < 2 && students[i + 1] == 2) {
                students[i + 1] -= 1
                students[i] += 1
            }
        }
    }
    
    for i in 1...n {
        if students[i] == 0 {
            if students[i - 1] == 2 {
                students[i - 1] -= 1
                students[i] += 1
                continue
            }
            
            if students[i + 1] == 2 {
                students[i + 1] -= 1
                students[i] += 1
            }
        }
    }
    
    for item in students {
        if item >= 1 {
            result += 1
        }
    }
    
    return result
}

File: Programmers/노수일/Q42862.py
def solution(n, lost, reserve):
    for i in range(len(lost)):
        for j in range(len(reserve)):
            if reserve[j] == lost[i]:
                lost[i] = 0
                reserve[j] = 0
                break

    for i in range(len(lost)):
        if lost[i] != 0:
            for j in range(len(reserve)):
                if reserve[j] != 0:
                    if reserve[j] - 1 == lost[i] or reserve[j] + 1 == lost[i]:
                        lost[i] = 0
                        reserve[j] = 0
                        break

    answer = n
    for i in range(len(lost)):
        if lost[i] != 0:
            answer -= 1

    return answer


print(solution(5, [1, 2, 3, 4, 5], [1]))
print(solution(5, [2, 4], [1, 3, 5]))
print(solution(5, [2, 4], [3]))
print(solution(3, [3], [1]))
